### DISCUSSION TITLE
How to change LXQt window title bar

### DISCUSSION THREAD

**Discussion by edwardp3** [lxqt/lxqt#2356] [Q&A] [answered]
LXQt 1.2.0, Debian sid/unstable. Subsequently installed obconf-qt.

I would like to change the window title bar from the default gray to one installed that works with LXDE, Mistral, but the Openbox Settings window (its menu entry appeared upon install of the `obconf-qt`package), does not change the theme, Clearlooks was highlighted by default, but its sample graphics which are visible in the LXDE Openbox Configuration Manager, do not match what was displayed in LXQt. Selecting _any_ theme doesn't change the window title bar.

Could the LXQt install possibly be missing a package, or is it even possible to change the window title bar?

Thanks in advance.

**Comment by stefonarch** [comment, marked as answer]
Your probably not using openbox but xfwm4, if I remember well there is not much to do about styling it. You can switch to openbox in session settings.

**Reply by edwardp3** [reply]
Switching it to openbox, allowed (what I should have called it) the Openbox Theme to be changed.

Thank you.

**Reply by edwardp3** [reply]
Also want to add that when I tried picom again, it didn't display any window title bars. Switching back to xfwm4 or openbox, restored them.

**Reply by tsujan** [reply]
The title-bar is drawn by the window manager (WM). picom is a compositor, not a WM. A compositor allows you to have translucent windows (like panel or qterminal), window shadows, etc.

picom needs a WM (like openbox) running. xfwm4 is both a WM and a compositor.

**Reply by stefonarch** [reply]
One _can_ use picom + xfwm4 when it's compositor is disabled.
I used this for a while. 

>  it didn't display any window title bars

strange, maybe invisible?

**Reply by edwardp3** [reply]
When I moused over where the window title bar would have been, nothing occurred, I'm guessing they simply did not appear.

After I installed LXQt on Debian (by installing the `task-lxqt-desktop` package), I noticed afterwards that it also installed `kwin`. Both `kwin` and `kwin_x11` are now listed as window managers and on the login screen, there is now an entry for `Plasma (X11)`.

I am using `kwin_x11` for this session. The interface (and in typing this) is noticeably slower (and will occasionally be jittery when minimizing/closing) compared to using xfwm4 and openbox, but I like the fade-in of the menu and the minimizing/maximizing of the window effects. `kwin` provides an option to change the window title bar/theming.

**Comment by isf63** [comment]
Check for a "xfwm4-themes" package in your distro's repos. It contains ~40-50 additional themes. This package exists on Ubuntu, openSUSE, Arch Linux, Fedora, and possibly others.

**Reply by edwardp3** [reply]
Debian has `xfwm4-theme-breeze` in their repo and it is already installed. There are no other xfwm4 theme packages in the repo.

**Reply by isf63** [reply]
There exists the .deb file for Ubuntu. http://ports.ubuntu.com/pool/universe/x/xfwm4-themes/xfwm4-themes_4.10.0-2_all.deb On Debian I believe you can do `sudo apt install <.deb>`.

**Comment by eriefisher** [comment]
I realize this is an aging post but...... in Siduction they offer lxqt-xfwm-helper. It's just the .desktop files to start "window manager settings" and "window manager tweaks" which is already installed if you have xfwm4 installed. This gives you easy access to window decorations and the compositor settings etc. 

Maybe this is something that can be reworked to fit well into LXQT.

**Reply by Gnomasz** [reply]
I'm also late to the party, but better late then never, hopefully.
> Maybe this is something that can be reworked to fit well into LXQT.

I think it's against the current principle of not supporting any WM by the LXQt team, and ceding it on the distribution developers. And as much as I was dismayed by the retirement of obconf-qt, I understand that the team's not that numerous and needs to focus - and it seems to me that the development of LXQt picked up the pace after Openbox was left behind (though it might be just an impression, I haven't looked at the data).

It makes me think, though, that the [wiki page on WM integration](https://github.com/lxqt/lxqt/wiki/ConfigWindowManagers) could use links to tools like lxqt-xfwm-helper (if there is one, as I couldn't find any) or [lxqt-desktop-shell plugin](https://gitlab.com/wayfireplugins/lxqt-desktop-shell.git), and the page could be linked on the [Third party apps](https://github.com/lxqt/lxqt/wiki/Third-party-apps) page. The thing is, for the wiki to really work, the authors or maybe users of such tools would need to contribute.

**Reply by stefonarch** [reply]
For `lxqt-desktop-shell` it requires both wayfire and plugin compilation from sources and there are some  issues in both not solved yet. In my opinion the wiki should  not propose tools or applications which are not in the repositories of the the main distros.  Advanced user will find their way.
